packages/client/src/modals/SuccessModal.js

```diff
@@ -0,0 +1,140 @@
+import React, { useContext, useEffect, useState } from "react";
+import { Web3Context } from "contexts/Web3";
+import { useClipboard } from "../hooks";
+import { Copy, ArrowUp, OpenNewTab, Close } from "../components/Svg";
+import { getFlowscanUrlForTransaction } from "utils";
+import { ASSET_TYPES } from "../constants/enums";
+import { COIN_TYPE_TO_META, CONTRACT_NAME_TO_COIN_TYPE } from "../constants/maps";
+
+const SuccessModal = ({ actionData, actionType, txID, onClose, safeName, safeAddress, safeOwners }) => {
+    const { getNFTReference } = useContext(Web3Context);
+    const clipboard = useClipboard();
+
+    const { recipientAddr, nftID, collectionID, vaultID, amount } = actionData;
+    const { displayName, icon } = getTokenMeta(vaultID) || {};
+
+    const [image, setImage] = useState();
+    const { name: imageName, imageURI } = image || {};
+
+    useEffect(() => {
+        if (actionType === ASSET_TYPES.NFT) {
+            const getImageURL = async () => {
+                const result = await getNFTReference(recipientAddr, nftID);
+                setImage(result);
+            };
+            getImageURL().catch(console.error);
+        }
+    }, [actionType, nftID, recipientAddr, getNFTReference])
+
+    function getNFTName(collectionId) {
+        return collectionId.split(".")[2];
+    }
+
+    function getTokenMeta(vaultId) {
+        if (vaultId) {
+            const tokenName = vaultId.split(".")[2];
+            return COIN_TYPE_TO_META[CONTRACT_NAME_TO_COIN_TYPE[tokenName]];
+        }
+    }
+
+    function getSafeOwnerName(address) {
+        const owner = safeOwners.filter(owner => owner.address === address);
```

Comment:
suggest using safeOwners.find() it will return the first encounter of qualified element, so we don't have to get it at index 0 later in the code

Reply:
Updated